pkg/registry/batch/job/strategy.go

```diff
@@ -191,10 +192,27 @@ func validationOptionsForJob(newJob, oldJob *batch.Job) batchvalidation.JobValid
 		_, hadJobName := oldJob.Spec.Template.Labels[batch.JobNameLabel]
 		_, hadControllerUid := oldJob.Spec.Template.Labels[batch.ControllerUidLabel]
 		opts.RequirePrefixedLabels = hadJobName && hadControllerUid
+
+		if oldJob.Spec.PodFailurePolicy != nil {
+			opts.AllowZeroInPodFailurePolicy = IsZeroUsedInPodFailurePolicy(oldJob.Spec.PodFailurePolicy)
+		}
 	}
 	return opts
 }
 
+func IsZeroUsedInPodFailurePolicy(podFailurePolicy *batch.PodFailurePolicy) bool {
```

Comment:
We can check if podFailurePolicy is nil in this function. This will eliminate the need to check it outside in two places.